packages/cyberstorm/src/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { Fragment } from "react";
+import styles from "./componentStyles/Breadcrumb.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faHouse, faSlashForward } from "@fortawesome/pro-regular-svg-icons";
+import { LinkingProvider } from "@thunderstore/components";
+import { LinkLibrary } from "./LinkLibrary";
+
+interface Crumb {
+  label: string;
+  LinkComponent?: (props: any) => React.ReactElement | null; // eslint-disable-line @typescript-eslint/no-explicit-any
+  LinkProps?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+interface Props {
+  parts: Crumb[];
+}
+
+/**
+ * Navigational bread crumbs defining where the user is at the site.
+ */
+export const BreadCrumb: React.FC<Props> = (props) => {
+  if (!props.parts) {
+    return (
+      <div className={`${styles.breadcrumb}`}>
+        <Home />
+      </div>
+    );
+  }
+  const parts = [...props.parts];
+  const leaf = parts.pop();
+
+  return (
+    <div className={`${styles.breadcrumb}`}>
+      <Home />
+      {(parts.length || leaf !== undefined) && <Separator />}
+
+      {parts.map((crumb, i) => (
+        <Fragment key={`${crumb.label}-${i}`}>
+          <MidCrumb crumb={crumb} />
+          <Separator />
+        </Fragment>
+      ))}
+
+      <LeafCrumb crumb={leaf} />
+    </div>
+  );
+};
+
+const Home: React.FC = () => {
+  return (
+    <LinkingProvider value={LinkLibrary}>
```

Comment:
LinkLibrary shouldn't be part of `@thunderstore/cyberstorm` (more about this in a bit) so we can't really hard code the link here. And I'm not sure we even want to hard code it to point to index? Maybe we want to give the apps a possibility to point the root node to e.g. community listing instead of landing page with "download the MM" hype?

I like the idea of not having to pass the home node everywhere this is going to be used, but it just doesn't work that well in this particular case.

Reply:
I've edited this to work so that, if no parts are passed, it will include `DefaultHomeCrumb` part in the breadcrumb. So the "home" isn't forced, but it is assured that there is something always in the breadcrumb.

Reply:
To me it seems L25-27 returns null if no parts are passed, and `DefaultHomeCrumb` is never used anywhere (which you're IDE would probably warn about, except you've exported it for some reason, which I think you shouldn't do either).

Reply:
DefaultHomeCrumb is used within the .stories file to add it as the first part in the different templates. I didn't want to use it in the Breadcrumb component by default, cause we might not want it to be the first crumb. (IIRC from a previous conversation somewhere)

Reply:
> DefaultHomeCrumb is used within the .stories file

No, it's not.